pkg/druid.go

```diff
@@ -84,6 +87,58 @@ func newDataSourceInstance(settings backend.DataSourceInstanceSettings) (instanc
 	if basicAuth := data.Get("connection.basicAuth").MustBool(); basicAuth {
 		druidOpts = append(druidOpts, druid.WithBasicAuth(data.Get("connection.basicAuthUser").MustString(), secureData["connection.basicAuthPassword"]))
 	}
+	if mTLS := data.Get("connection.mTLS").MustBool(); mTLS {
+		log.DefaultLogger.Info("mTLS enabled for Druid connection")
+
+		cert, ok := secureData["connection.mTLSCert"]
+		if !ok || cert == "" {
+			return &druidInstanceSettings{}, fmt.Errorf("mTLS certificate is required but not provided")
+		}
+		key, ok := secureData["connection.mTLSKey"]
+		if !ok || key == "" {
+			return &druidInstanceSettings{}, fmt.Errorf("mTLS key is required but not provided")
+		}
+		ca, hasCA := secureData["connection.mTLSCa"]
+
+		useSystemCAPool := data.Get("connection.mTLSUseSystemCaPool").MustBool()
+
+		clientCert, err := tls.X509KeyPair([]byte(cert), []byte(key))
+		if err != nil {
+			return &druidInstanceSettings{}, fmt.Errorf("failed to load client certificate and key: %w", err)
+		}
+
+		caCertPool := x509.NewCertPool()
+		if useSystemCAPool {
+			caCertPool, err = x509.SystemCertPool()
+			if err != nil {
+				return &druidInstanceSettings{}, fmt.Errorf("failed to load system CA pool: %w", err)
+			}
```

Comment:
Do you think it would be better to return an error here, or should we fall back to creating an empty pool if the operation fails?

Reply:
I personally prefer to be as explicit with the user as possible to not hide potential issues they may run into. I'd hate for a user to say "use the system ca pool" and the datasource just quietly does not.